### PR TITLE
Added option for configuration file

### PR DESCRIPTION
**Pull request by mjavier2k**
- added option to use config.yaml
- updated README.md

### CODE REVIEW

**Comment by mjavier2k** [comment]
implements https://github.com/mjavier2k/solidfire-exporter/issues/25

**Comment by mjavier2k** [comment]
When **viper.ReadInConfig()** is called, it reads the config file which has the same key as whats on https://github.com/mjavier2k/solidfire-exporter/blob/master/pkg/solidfire/types.go#L8-L14

that becomes the default values. that can be overriden by either flags or env.

**Comment by neufeldtech** [comment]
What I meant is that we're not doing any 'reading' of the viper config anywhere after it exists, i.e. `viper.Get("endpoint")`
We also need to consider the order of operations in terms of if flags will override the config file, etc. Viper could provide some help with this if we adopt the viper-native approach. If we're making breaking changes to the configuration anyway (by removing the option of passing username/password on the CLI via a flag), then we might as well adopt the viper-native approach where it can automatically 'detect' environment variables based on our go variable names, and automatically handle config file and flag order of operations.

**Comment by neufeldtech** [comment]
My Bad - I forgot that we are _already_ using viper, so your config file pattern just magically works
https://github.com/mjavier2k/solidfire-exporter/blob/ca5e82552b8c6ecd5337482409b6f538f85868da/pkg/solidfire/solidfire.go#L34-L36

**Comment by neufeldtech** [comment]
Do we want to drop support for passing in username/password as flags?

**Comment by mjavier2k** [comment]
Yeah, lets drop that support. i'll remove the username/password option on the command line flag

**Comment by neufeldtech** [comment]
Is there a need for a config flag for the config file itself?
I.e. 
```
-c path to config.yaml file
```

File: cmd/solidfire-exporter/main.go [hunk 1]
```diff
@@ -36,12 +36,26 @@ func init() {
 	} else {
 		viper.BindEnv(solidfire.ListenPortFlag, solidfire.ListenPortFlagEnv)
 	}
-
-	viper.BindEnv(solidfire.UsernameFlag, solidfire.UsernameFlagEnv)
```
Comment (neufeldtech) [inline cmd/solidfire-exporter/main.go:40]: If we remove these here, can we still pass username/password through an environment variable?
As I recall, we want to support username and password via an Environment variable or config file, but want to drop support for having them as command-line flags.

File: pkg/solidfire/types.go [hunk 2]
```diff
@@ -12,9 +12,10 @@ var (
 	EndpointFlag          = "endpoint"
 	InsecureSSLFlag       = "insecure"
 	HTTPClientTimeoutFlag = "timeout"
-	ListenPortFlagEnv     = "SOLIDFIRE_PORT"
+	ConfigFileFlag        = "config.yaml"
```
Comment (neufeldtech) [inline pkg/solidfire/types.go:15]: This is the flag _name_ not the flag's default value.

You probably meant this to be as such:

```suggestion
	ConfigFileFlag        = "config"
```

**Pull request closed without merging**